### ISSUE TITLE
PSBT: is finalized state queryable?

### ISSUE THREAD

**Issue by AdamISZ** [Simplexum/python-bitcointx#44]
On a brief look through the code today, I can't see functions/flags that let a caller check if the state of a given `PSBT_Input` is finalized or not (or the PSBT as a whole). I know I can check `final_script_sig` and `final_script_witness` but it seems like it might be useful.

**Comment by dgpv** [comment]
I do it like this:

```
    def extract_transaction(self) -> CTransaction:
        sign_result = self.sign(KeyStore())
        if not sign_result.is_final:
            raise ValueError(
                'not all inputs have required signatures')
```

This also does the consistency checks along the way.

It may be a bit wasteful if you only want to know if `final_*` fields were set, but adding `is_final()` method that would only do `self.final_script_witness or self.final_script_sig` is not in the spirit of the library ("consistency where possible/reasonable").

Do you thinkg a convenience `is_final()` that does `self.sign(KeyStore()).is_final()` would be useful ?

**Comment by dgpv** [comment]
Maybe take the `final_script_witness` and `final_script_sig` check logic from `sign()` into `is_final()` and call `is_final()` from sign.

**Comment by dgpv** [comment]
Check out #46 (untested at the moment), what do you think ?

**Comment by dgpv** [comment]
`is_final()` will be included in 1.1.1.post1, along with a fix to restore expected `set_utxo(None, ...)` semantics.

#46 passes tests, but could benefit from some review, nevertheless.
